### PR TITLE
DIGI_10_createAddOn

### PR DESCRIPTION
**Pull request by bommavikas**
1)Created AddOnController for Create AddOn
2)Created AddOnLibrary
3)Created View for Create AddOn
4)Created AddOnModel
5)Modified LibraryUtility

### CODE REVIEW

File: DigiOrderEnterprise/DigiOrderEnterprise/Web.config [hunk 1]
```diff
@@ -12,7 +12,6 @@
 	  <add key="digiapiurl" value="http://localhost/DigiOrderApi/api/" />
 	  <add key="digiapiuserid" value="ENTERPRISE" />
 	  <add key="digiapipassword" value="B75A5773-E01A-422C-9958-6194D8E9D0F8"/>
-    <add key="digiapigrant_type" value="password"/>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Web.config:15]: add password to config file

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj [hunk 2]
```diff
@@ -34,9 +34,6 @@
     <Reference Include="DigiOrderEnterpriseModel">
       <HintPath>..\DigiOrderEnterpriseModel\bin\Debug\DigiOrderEnterpriseModel.dll</HintPath>
     </Reference>
-    <Reference Include="Newtonsoft.Json, Version=9.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
-      <HintPath>..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
-    </Reference>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:39]: add these reference

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj [hunk 3]
```diff
@@ -52,10 +49,6 @@
     <Compile Include="EndPoint.cs" />
     <Compile Include="LibraryUtility.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
-    <Compile Include="TaxPageLibrary.cs" />
-  </ItemGroup>
-  <ItemGroup>
-    <None Include="packages.config" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:58]: Why you removed taxpagelibrary

File: DigiOrderEnterprise/DigiOrderEnterprise/Web.config (hunk 1 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Web.config:15]: added

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 4]
```diff
@@ -45,7 +45,7 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="DigiOrderEnterpriseLibrary, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
+  <Reference Include="DigiOrderEnterpriseLibrary, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:48]: Add space before reference

File: DigiOrderEnterprise/DigiOrderEnterprise/Web.config [hunk 5]
```diff
@@ -9,7 +9,7 @@
       <add key="webpages:Enabled" value="false" />
       <add key="ClientValidationEnabled" value="true" />
       <add key="UnobtrusiveJavaScriptEnabled" value="true" />
-	  <add key="digiapiurl" value="http://localhost/DigiOrderApi/api/" />
+	  <add key="digiapiurl" value="http://localhost:52893/api/" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Web.config:12]: Remove port address

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj [hunk 6]
```diff
@@ -51,11 +51,26 @@
     <Compile Include="CategoryLibrary.cs" />
     <Compile Include="EndPoint.cs" />
     <Compile Include="LibraryUtility.cs" />
+    <Compile Include="obj\Debug\TemporaryGeneratedFile_036C0B5B-1481-4323-8D20-8F5ADCB23D92.cs" />
+    <Compile Include="obj\Debug\TemporaryGeneratedFile_5937a670-0e60-4077-877b-f7221da3dda1.cs" />
+    <Compile Include="obj\Debug\TemporaryGeneratedFile_E7A71F73-0F8D-4B9B-B56E-8E70B10BC5D3.cs" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:56]: Remove these debug file

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj [hunk 7]
```diff
@@ -51,11 +51,26 @@
...
     <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="AddOnLibrary.cs" />
     <Compile Include="TaxPageLibrary.cs" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="obj\Debug\build.force" />
+    <None Include="obj\Debug\DesignTimeResolveAssemblyReferencesInput.cache" />
+    <None Include="obj\Debug\DigiOrderEnterpriseLibrary.csproj.CoreCompileInputs.cache" />
+    <None Include="obj\Debug\DigiOrderEnterpriseLibrary.csprojAssemblyReference.cache" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:65]: Remove these debug files

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj [hunk 8]
```diff
@@ -51,11 +51,26 @@
...
     <None Include="packages.config" />
   </ItemGroup>
+  <ItemGroup>
+    <Content Include="obj\Debug\DigiOrderEnterpriseLibrary.csproj.FileListAbsolute.txt" />
+  </ItemGroup>
+  <ItemGroup>
+    <Folder Include="bin\Debug\" />
+    <Folder Include="obj\Debug\TempPE\" />
+  </ItemGroup>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:74]: Remove these debug files

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj (hunk 3 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj]: Included TaxPageLibrary

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Mvc;
+
+
+namespace DigiOrderEnterprise.Controllers
+{
+    public class AddOnController : Controller
+    {
+        // GET: MenuItem
```
[the remark this answers is not in the document]
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs]: REMOVED

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            ViewBag.ControlNviation = "CreateAddOn";
+            return View(new AddOnModel());
+        }
+
+        /// Author: Vikas Bomma
+        /// Date: 7/14/2020
+        /// Description: Inserting AddOns To database
+        /// Ref. StoryId: DIGI 10
+        /// Reviewer:
```
[the remark this answers is not in the document]
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs]: Added

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj (hunk 2 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj]: Added

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+                Task<HttpResponseMessage> res = addOnLibrary.createAddOn(addOnModel);
+                if (res.IsFaulted)
+                {
+                    TempData["fmessage"] = "Insertion Of AddOn Failed";
+                    return RedirectToAction("Create");
+                }
+            }
+            catch (Exception ex)
+            {
+                ViewBag.ErrorMessage = "";
```
[the remark this answers is not in the document]
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs]: Added

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 4 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Added

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj (hunk 6 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj]: Removed

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj (hunk 8 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj]: Removed

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 12]
```diff
@@ -235,7 +236,9 @@
     <Content Include="asset\fonts\glyphicons-halflings-regular.ttf" />
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff" />
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff2" />
-    <Content Include="Web.config" />
+    <Content Include="Web.config">
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:239]: check  spaces in line 239 compare with removed one

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 13]
```diff
@@ -235,7 +236,9 @@
...
+      <SubType>Designer</SubType>
+    </Content>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:241]: Remove these two lines

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/AddOnLibrary.cs [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace DigiOrderEnterpriseLibrary
+{
+    public class AddOnLibrary
+    {
+        LibraryUtility libraryUtility = null;
+        Task<HttpResponseMessage> httpResponse = null;
+
+        public Task<HttpResponseMessage> createAddOn(string TYPE, string NAME, string DETAILS, double? PRICE)
+        {
```
[the remark this answers is not in the document]
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/AddOnLibrary.cs:16]: Taken all Values In Model Object

File: DigiOrderEnterprise/DigiOrderEnterprise/Web.config (hunk 5 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Web.config]: Removed Port Adreess And Replaced

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj (hunk 7 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj]: Removed Debug Files

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 13 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Removed Lines

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 15]
```diff
@@ -235,7 +236,7 @@
     <Content Include="asset\fonts\glyphicons-halflings-regular.ttf" />
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff" />
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff2" />
-    <Content Include="Web.config" />
+ <Content Include="Web.config" />
     <Content Include="Web.Debug.config">
       <DependentUpon>Web.config</DependentUpon>
     </Content>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:242]: Revert this file

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj [hunk 16]
```diff
@@ -52,10 +52,11 @@
     <Compile Include="EndPoint.cs" />
     <Compile Include="LibraryUtility.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="AddOnLibrary.cs" />
     <Compile Include="TaxPageLibrary.cs" />
   </ItemGroup>
   <ItemGroup>
     <None Include="packages.config" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:59]: Revert this file

File: DigiOrderEnterprise/DigiOrderEnterpriseModel/DigiOrderEnterpriseModel.csproj [hunk 17]
```diff
@@ -43,6 +43,7 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="AddOnModel.cs" />
     <Compile Include="CategoryModel.cs" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseModel/DigiOrderEnterpriseModel.csproj:47]: Revert this file

File: DigiOrderEnterprise/DigiOrderEnterpriseModel/AddOnModel.cs [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+        public string NAME { get; set; }
+        [Required]
+        public string DETAILS { get; set; }
+        public double? PRICE { get; set; }
+        public string IMAGE { get; set; }
+        public bool STATUS { get; set; }
+        public string Dated { get; set; }
+        public string Updated { get; set; }
+    }
+}
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseModel/AddOnModel.cs:22]: Revert this file as you have checkined in list addons pr

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 15 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:242]: Reverted

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj (hunk 16 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj]: Reveretd

File: DigiOrderEnterprise/DigiOrderEnterpriseModel/DigiOrderEnterpriseModel.csproj (hunk 17 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseModel/DigiOrderEnterpriseModel.csproj]: Reverted

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 19]
```diff
@@ -239,6 +240,7 @@
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff2" />
     <Content Include="Views\TaxParameter\Index.cshtml" />
     <Content Include="Web.config" />
+ <Content Include="Web.config" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:243]: Remove this

**Review by anebhargavkrishna — CHANGES_REQUESTED** [review]

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 19 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Removed

**Pull request merged**